lib/usagereporter/teleport/aggregating/reporter.go

```diff
@@ -201,9 +203,9 @@ func (r *Reporter) run(ctx context.Context) {
 	ticker := r.clock.NewTicker(checkInterval)
 	defer ticker.Stop()
 
-	startTime := r.clock.Now().UTC().Truncate(reportGranularity)
-	windowStart := startTime.Add(-rollbackGrace)
-	windowEnd := startTime.Add(reportGranularity)
+	userActivityStartTime := r.clock.Now().UTC().Truncate(userActivityReportGranularity)
```

Comment:
Can we please take this opportunity to provide a decent comment outlining what this function does? It's quite a bit of non-trivial code.

Reply:
Tried to do it in https://github.com/gravitational/teleport/pull/34954/commits/701adf85e0e143e40d558a7f6e293b590a2ebbb0